### PR TITLE
Fix `TextInput` background color style when hovered

### PR DESCRIPTION
**Pull request by CarlosCortizasCT**
#### Summary

Fix `TextInput` background color style when hovered

## Description

It seems we introduced a regression bug when working in `TextInput` adjustments to support the new theme.
When hovering the component, its background color is changed to transparent so, if the component is in a container with any background color, the input itself changes its background color as well.

This is an example of this behaviour:
Components _ Inputs - TextInput ⋅ Storybook (1)

This is the behaviour after applying the fix:
Components _ Inputs - TextInput ⋅ Storybook (2)

### CODE REVIEW

File: design-system/materials/internals/definition.yaml [hunk 1]
```diff
@@ -387,10 +387,10 @@ decisionGroupsByTheme:
           choice: color-accent-98
         background-color-for-input-when-hovered:
           description: ''
-          choice: 'unset'
+          choice: color-surface
         background-color-for-input-when-focused:
           description: ''
-          choice: 'unset'
+          choice: color-surface
```
Comment (CarlosCortizasCT) [inline design-system/materials/internals/definition.yaml:393]: This is the actual fix

File: docs/.storybook/decorators/section/section.js [hunk 2]
```diff
@@ -15,6 +15,9 @@ const Section = (props) => {
   );
 };
 
-Section.propTypes = { children: PropTypes.node.isRequired };
+Section.propTypes = {
```
Comment (CarlosCortizasCT) [inline docs/.storybook/decorators/section/section.js:18]: This is to allow changing the background color for stories containers.

File: packages/components/inputs/text-input/src/text-input.story.js [hunk 3]
```diff
@@ -17,7 +17,16 @@ storiesOf('Components|Inputs', module)
   })
   .add('TextInput', () => {
     return (
-      <Section>
+      <Section
```
Comment (CarlosCortizasCT) [inline packages/components/inputs/text-input/src/text-input.story.js:20]: This allows to change the background color of the story container to check the correct behaviour.

**Review by tdeekens — APPROVED** [review]

**Review by Rhotimee — APPROVED** [review]
💯

**Pull request merged**